### ISSUE TITLE
Errors in example file, caused by use of gcc

### ISSUE THREAD

**Issue by Yoekkul** [DeadCodeProductions/diopter#92]
Running the example program in `examples/reduced_program_with_O3_smaller_than_Os.py` returns the following error.
```
O3 size: 255
Os size: 274

C-Reduce cannot run because the interestingness test does not return
zero. Please ensure that it does so not only in the directory where
you are invoking C-Reduce, but also in an arbitrary temporary
directory containing only the files that are being reduced. In other
words, running these commands:

  DIR=`mktemp -d`
  cp /tmp/tmpcgskp9k1/code.c $DIR
  cd $DIR
  /tmp/tmpcgskp9k1/check.py
  echo $?

should result in "0" being echoed to the terminal.

See "creduce --help" for more information.

Traceback (most recent call last):
  File "/home/diopter/examples/reduced_program_with_O3_smaller_than_Os.py", line 71, in <module>
    assert rprogram
AssertionError
```

This seems to be caused by the preprocessing step `[65]     p = Os.preprocess_program(p, make_compiler_agnostic=True)`, which always makes the sanitize test in the test function fail.  
It gets fixed when I change the compiler from gcc to clang, in the following way:
```
 O3 = CompilationSetting(
        compiler=CompilerExe.get_system_clang(),
        opt_level=OptLevel.O3,
        flags=("-march=native",),
    )
    Os = CompilationSetting(
        compiler=CompilerExe.get_system_clang(),
        opt_level=OptLevel.Os,
        flags=("-march=native",),
    )
```

**Comment by thetheodor** [comment]
@Yoekkul thanks for the bug report. I can't reproduce this locally, probably due to different gcc versions. What is the output of `gcc -v` on the the system you run this?

**Comment by thetheodor** [comment]
Fixed with #95
